polynote-kernel/src/main/scala/polynote/kernel/Kernel.scala

```diff
@@ -32,7 +32,12 @@ trait Kernel {
   /**
     * Provide a location for the definition of whatever is at the given position in the given cell
     */
-  def goToDefinition(fileOrCell: Either[String, CellID], pos: Int): TaskC[(List[DefinitionLocation], Option[String])]
+  def goToDefinition(fileOrCell: Either[String, CellID], pos: Int): TaskC[List[DefinitionLocation]]
+
+  /**
+    * Retrieve the source of the given dependency identifier from the given language
+    */
+  def dependencySource(language: String, dependency: String): RIO[BaseEnv with GlobalEnv, String]
```

Comment:
Split this back up, so we separately get the definition locations, and then can ask for the source code in a different request. The dependency sources are now handled differently – they're requested in a separate HTTP request instead of over the websocket.

Reply:
What motivated the change?

Reply:
Mainly, it's how Monaco wants it to be